server/apps/immich/src/config/asset-upload.config.ts

```diff
@@ -20,13 +20,18 @@ export const assetUploadOption: MulterOptions = {
   storage: diskStorage({
     destination: (req: Request, file: Express.Multer.File, cb: any) => {
       const basePath = APP_UPLOAD_LOCATION;
-      const fileInfo = req.body as CreateAssetDto;
+      // TODO these are currently not used. Shall we remove them?
+      // const fileInfo = req.body as CreateAssetDto;
 
-      const yearInfo = new Date(fileInfo.createdAt).getFullYear();
-      const monthInfo = new Date(fileInfo.createdAt).getMonth();
+      // const yearInfo = new Date(fileInfo.createdAt).getFullYear();
+      // const monthInfo = new Date(fileInfo.createdAt).getMonth();
+
+      if (!req.user) {
+        return;
+      }
 
       if (file.fieldname == 'assetData') {
-        const originalUploadFolder = `${basePath}/${req.user['id']}/original/${req.body['deviceId']}`;
+        const originalUploadFolder = `${basePath}/${req.user.id}/original/${req.body['deviceId']}`;
```

Comment:
So I got this error when trying to run the app

<img width="879" alt="Screen Shot 2022-06-24 at 22 01 46" src="https://user-images.githubusercontent.com/27055614/175755787-1a99d338-0230-45ed-b91b-06b328b7ea82.png">

And the type of `user` here is `User` from `Express.User`
<img width="862" alt="Screen Shot 2022-06-24 at 22 01 40" src="https://user-images.githubusercontent.com/27055614/175755796-61b75e40-3ab3-463b-8bac-2c2d321ac2d0.png">

Reply:
The user here is returned from the function `validate` in `jwt.strategy.ts` which is a UserEntit
<img width="630" alt="Screen Shot 2022-06-24 at 22 18 19" src="https://user-images.githubusercontent.com/27055614/175756182-ce8c9b18-880e-4134-a0a2-a24fd2b1695e.png">
y

Reply:
Here is my proposed fix for this issue in `asset-upload.config.ts` and `profile-image-upload.config.ts`

<img width="878" alt="image" src="https://user-images.githubusercontent.com/27055614/175756265-e0582470-89a1-4eb4-a912-7613eee16c49.png">

Reply:
Yes sorry about that, the e2e test was working in my computer (this is also the reason why e2e tests failed in CI) because the container was already created and still  had the old `tsconfig` inside, so no strict mode 😅.

I have added a much neater fix, but would probably need to move the `global.d.ts` somewhere inside src, I'll take a look at fixing this.
Using `as` should be the last resort in TypeScript (also `any` but this is a subject for another day 😄)

I had added a short explanation in the git commit message:
```
override Express/User interface to use UserResponseDto interface
     This is for when the accessing the `user` from a Express Request,
     like in `asset-upload-config`
```

So Express left the `User` interface empty, so you can override it with your own interface like
```
declare global {
  namespace Express {
    // eslint-disable-next-line @typescript-eslint/no-empty-interface
    interface User extends UserResponseDto {}
  }
}
```
I've used `UserResponseDto` for now, although we can create a specific interface for the user returned in `jwt.strategy`